lib/Sema/TypeCheckType.cpp

```diff
@@ -4975,8 +4975,29 @@ TypeResolver::resolveSendingTypeRepr(SendingTypeRepr *repr,
     return ErrorType::get(getASTContext());
   }
 
+  NeverNullType resolvedType = resolveType(repr->getBase(), options);
+  
+  // If resolved type is Sendable, warn about unnecessary 'sending' annotation.
+  if (options.is(TypeResolverContext::FunctionInput)) {
+    if(!resolvedType->hasTypeParameter()) {
+      if (resolvedType->isSendableType()) {
+        diagnose(repr->getSpecifierLoc(),
+                 diag::sending_applied_to_sendable,
+                 resolvedType.get())
+        .fixItRemove(repr->getSpecifierLoc());
+      }
+    } else {
+      if (resolvedType->getASTContext().getProtocol(KnownProtocolKind::Sendable)) {
```

Comment:
This is only getting the known protocol declaration from the ASTContext. Are you trying to check whether a type with a type parameter is conforming to `Sendable` here?